README.md

```diff
@@ -16,9 +16,40 @@ In summary, our software suite addresses the multifaceted challenges associated
+
+Power users of NWBInspector often seek more direct control of the inspection process and its outputs. To enable more flexibility when using _NWBInspector_, we provide a simple Python API with all the same functionality as the CLI, along with some more advanced features. Just like the CLI, the Python API can be employed to inspect individual NWB files and entire directories of NWB files with the `nwbinspector.inspect_nwbfile()` function and `nwbinspector.inspect_all()` function respectively. Additionally, the Python API allows users to inspect NWB files directly in memory with the `nwbinspector.inspect_nwbfile_object()` function. Since both the `inspect_nwbfile` and the `inspect_all` functions return generators, Python users can employ these functions in more sophisticated ways, such as implementing early stopping if a particular message is raised. These features empower advanced users to fully leverage _NWBInspector_.
+
+Some organizations have their members use NWBInspector, but they may need to adjust the importance level of some checks to meet their requirements. For instance, DANDI requires users to specify the sex and age of their subjects, so those checks should be elevated to importance level "critical" instead of "best practice suggestions." To accommodate this need, NWBInspector provides a configuration option where the importance level of individual checks can be changed with a simple YAML file. Similarly, adding new checks is as simple as defining the check function in some module with a `register_check` decorator, and then specifying the module name with the `--modules` flag in the CLI. For example, in the module defining some new check:
+```python
+from nwbinspector.register_checks import available_checks, register_check, Importance
+
+@register_check(importance=Importance.SOME_IMPORTANCE_LEVEL, neurodata_type=some_neurodata_type)
+def check_personal_practice(...):
+    ...
+```
+and then in the command line, include that user-defined check,
+```bash
+    nwbinspector --modules path/to/module.py path/to/my/data.nwb
+```
+
+NWB has various Neurodata objects that all require appropriate checks, which are often the same or very similar across different objects. For instance, ROIResponseSeries, ElectricalSeries, and OptogeneticSeries all need to verify that their timestamps are non-negative and ascending. To avoid unnecessary duplication of checks, we leverage the inheritance of Neurodata objects in NWB and apply checks to all the children of a given Neurodata object. In this case, ROIResponseSeries, ElectricalSeries, and OptogeneticSeries are all descendants of TimeSeries. Thus, when we define `check_timestamps_ascending` for TimeSeries, it is automatically applied to all these objects. This approach allows NWBInspector to provide a minimal set of checks that leverages the inheritance structure of NWB to assess all the relevant objects.
```

Comment:
```suggestion
NWB has various neurodata objects that all require appropriate checks, which are often the same or very similar across different objects. For instance, ROIResponseSeries, ElectricalSeries, and OptogeneticSeries all need to verify that their timestamps are non-negative and ascending. To avoid unnecessary duplication of checks, we leverage the inheritance of Neurodata objects in NWB and apply checks to all the children of a given Neurodata object. In this case, ROIResponseSeries, ElectricalSeries, and OptogeneticSeries are all descendants of TimeSeries. Thus, when we define `check_timestamps_ascending` for TimeSeries, it is automatically applied to all these objects. This approach allows NWBInspector to provide a minimal set of checks that leverages the inheritance structure of NWB to assess all the relevant objects.
```